### PR TITLE
PhaseCallbacks.md

### PR DESCRIPTION
**Pull request by shaydeci**

### CODE REVIEW

File: documentation/source/PhaseCallbacks.md [hunk 1]
```diff
@@ -0,0 +1,280 @@
+# Phase Callbacks in SG
+
+Integrating your own code into an already existing training pipeline can draw much effort on the user's end.
+To tackle this challenge, a list of callables triggered at specific points of the training code can be passed through `phase_calbacks_list` inside `training_params` when calling `Trainer.train(...)`.
+
+SG's `super_gradients.training.utils.callbacks` module implements some common use cases as callbacks:
+
+    ModelConversionCheckCallback
+    DeciLabUploadCallback
```
Comment (ofrimasad) [inline documentation/source/PhaseCallbacks.md:9]: please remove DeciLabUploadCallback. it is only for "pro" users

File: documentation/source/PhaseCallbacks.md [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+      next_stage_start_epoch: 285
+...
+```
+
+Another example is how we use `BinarySegmentationVisualizationCallback` to visualize predictions during training in our [Segmentation Transfer Learning Notebook](https://bit.ly/3qKwMbe):
+
+
+## Integrating Your Code with Callbacks
+
+Integrating your code requires implementing a callback that `Trainer` would trigger in the equivalent areas inside SG's training pipeline.
```
Comment (ofrimasad) [inline documentation/source/PhaseCallbacks.md:48]: instead of "equivalent areas", "proper phases"

**Review by ofrimasad — COMMENTED** [review]
Beautiful tutorial. 
liked the example very much. 
had just two minor comments

File: documentation/source/PhaseCallbacks.md (hunk 2 above)
Comment (shaydeci) [reply documentation/source/PhaseCallbacks.md]: fixed.

File: documentation/source/PhaseCallbacks.md (hunk 1 above)
Comment (shaydeci) [reply documentation/source/PhaseCallbacks.md]: fixed

Comment (shaydeci) [reply documentation/source/PhaseCallbacks.md]: fixed in latest commit.

File: documentation/source/PhaseCallbacks.md (hunk 2 above)
Comment (shaydeci) [reply documentation/source/PhaseCallbacks.md]: fixed in latest commit.

**Review by ofrimasad — APPROVED** [review]
LGTM

**Pull request merged**